src/Integration/Rules/QualityProfileProviderCachingDecorator.cs

```diff
@@ -142,15 +150,25 @@ private void SynchronizeQualityProfiles()
         private bool isDisposed;
         public void Dispose()
         {
-            if (this.isDisposed)
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
I didn't think it was necessary to call _SuppressFinalize_ if the class doesn't have a finalizer, but our rule [S3881](https://rules.sonarsource.com/csharp/RSPEC-3881) complains if it isn't there.

Reply:
IIRC it is because the class is not sealed and child classes can have it.

Reply:
@michalb-sonar is right. We had one user who made us the remark that if the class is not sealed and doesn't have the `GC.SuppressFinalize` then sub-classes with a dtor wouldn't have the right behavior.